components/camel-jsonpath/src/test/java/org/apache/camel/jsonpath/JsonPathSplitSingleListTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.jsonpath;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.test.junit5.CamelTestSupport;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class JsonPathSplitSingleListTest extends CamelTestSupport {
+
+    @Override
+    protected RouteBuilder createRouteBuilder() {
+        return new RouteBuilder() {
+            @Override
+            public void configure() {
+                from("direct:start")
+                        // we select first book, but since we split after wards then ensure
+                        // it will be wrapped inside a List object.
+                        .split().jsonpath("$.store.book[0]", List.class)
+                        .to("mock:authors")
+                        .convertBodyTo(String.class);
+            }
+        };
+    }
+
+    @Test
+    public void testSplit() throws Exception {
```

Comment:
Can we add more tests to make sure the others patterns will work as expected? 
```
from("direct:start2")
                        // we select first book, but since we split after wards then ensure
                        // it will be wrapped inside a List object.
                        .split().jsonpath("$.store.book", List.class)
                        .to("mock:authors2")
                        .convertBodyTo(String.class);

                from("direct:start3")
                        // we select first book, but since we split after wards then ensure
                        // it will be wrapped inside a List object.
                        .split().jsonpath("$.store.book[*]", List.class)
                        .to("mock:authors3")
                        .convertBodyTo(String.class);
```

Reply:
Those 2 examples above will select all books and be a List already.
There is an online evaluator here: https://jsonpath.com/

Reply:
Yes, I'm suggesting it because in my tests with the 3.18.x base I use the `expr.setOptions(new Option[]{Option.ALWAYS_RETURN_LIST});` it resolve the $.store.book[0], but is breaking the $.store.book. Thus, I would like to make sure it will solve one scenario without changing the behavior of the other. Make sense?

Reply:
Yes I added your tests into class `JsonPathSplitMultipleListTest`

Reply:
And see `JsonPathSplitSingleListOptionTest` for an example with ALWAYS_RETURN_LIST

Reply:
Thank you! Can I backport to 3.18.x?

Reply:
No 3.18 is EOL. Upstream only 3.21 and 3.22 is supported.